### PR TITLE
Fix anonymous GitlabService

### LINKED ISSUE #481: GitLab service cannot be created without providing auth
ogr allows creating GitHub service instances without any authentication (provides read-only access)

Based on testing for #479 it's been found out that we cannot create GitLab instance without providing token (as we can with GitHub) *even though* it is supported by `python-gitlab` (https://python-gitlab.readthedocs.io/en/stable/api-usage.html#gitlab-gitlab-class)

### PR DESCRIPTION
**Pull request by mfocko**
- Allows no-auth read-only access to GitLab
  - authenticates only if provided token

Fixes #481

Signed-off-by: Matej Focko <mfocko@redhat.com>

### CODE REVIEW

**Review by lachmanfrantisek — COMMENTED** [review]
Nice!

What about some test so it's not so small change?..;)

**Comment by mfocko** [comment]
> What about some test so it's not so small change?..;)

lemme have a look

File: tests/integration/gitlab/test_service.py [hunk 1]
```diff
@@ -80,3 +82,7 @@ def test_project_create_in_the_group(self):
             repo=name_of_the_repo, namespace=namespace_of_the_repo
         )
         assert project.gitlab_repo
+
+    def test_service_without_auth(self):
+        service = GitlabService()
+        assert service.gitlab_instance
```
Comment (lachmanfrantisek) [inline tests/integration/gitlab/test_service.py:88]: Can you make some real call so we know it really works? It does not touch anything now...

Comment (mfocko) [reply tests/integration/gitlab/test_service.py:88]: Yes, I could... However I tested it with `master` and it fails with `401 Unauthorized` :D

**Comment by mfocko** [comment]
recheck

File: tests/integration/gitlab/test_service.py [hunk 2]
```diff
@@ -80,3 +82,8 @@ def test_project_create_in_the_group(self):
...
+        assert service.get_project(namespace="gnachman", repo="iterm2").exists()
```
Comment (lachmanfrantisek) [inline tests/integration/gitlab/test_service.py:89]: Why this?..;)

Comment (lachmanfrantisek) [reply tests/integration/gitlab/test_service.py:89]: Do we need to worry about the renaming of their repo? Or moving to `iterm2/iterm2` or something else?

**Review by lachmanfrantisek — APPROVED** [review]

File: tests/integration/gitlab/test_service.py (hunk 2 above)
Comment (mfocko) [reply tests/integration/gitlab/test_service.py:89]: they're already at version 3, so I wouldn't worry about it... I took it from the most stars and it was shorter than `inkscape/inkscape` :)

Comment (lachmanfrantisek) [reply tests/integration/gitlab/test_service.py:89]: ..;)

**Pull request merged**